frontend/src/App.tsx

```diff
@@ -1,19 +1,44 @@
-import { BrowserRouter as Router, Routes, Route } from 'react-router-dom';
+import {BrowserRouter as Router, Route, Routes} from 'react-router-dom';
 import {MainPage} from "./pages/MainPage.tsx";
 import Menu from "./components/Menu/Menu.tsx";
 import EditorPage from "./pages/EditorPage.tsx";
 import ResourcePage from "./pages/ResourcePage.tsx";
+import AuthCallbackPage from "./pages/AuthCallbackPage.tsx";
+import axios from 'axios';
+import {AuthProvider} from "./components/AuthProvider/AuthProvider.tsx";
+
+axios.interceptors.request.use(
+    (config) => {
+        try {
+            const token = localStorage.getItem('access_token');
+            if (token) {
+                config.headers.Authorization = `Bearer ${token}`;
+            }
+        } catch (error) {
+            console.warn('Cant attach bearer token:', error);
+        }
+        return config;
+    },
+    (error) => {
+        console.warn('Axios interceptor error:', error);
+        return Promise.resolve(error.config || {});
+    }
+);
```

Comment:
Can you separate this into another file? For example `axiosConfig.ts`.

Reply:
Lepszym podejściem jest przekazanie błędu dalej, aby można było go obsłużyć w miejscu wywołania żądania.

```suggestion
axios.interceptors.request.use(
  (config) => {
    const token = localStorage.getItem('access_token');
    if (token) {
      config.headers.Authorization = `Bearer ${token}`;
    }
    return config;
  },
  (error) => {
    console.error('Axios interceptor error:', error);
    return Promise.reject(error);
  }
);
```

Reply:
Separated to different file, but I didn't use your code.

Reply:
Fair